lib/explorer/series.ex

```diff
@@ -919,16 +919,20 @@ defmodule Explorer.Series do
   """
   @doc type: :element_wise
   @spec cast(series :: Series.t(), dtype :: dtype()) :: Series.t()
-  def cast(%Series{dtype: original_dtype} = series, dtype) do
-    if normalised = Shared.normalise_dtype(dtype) do
-      if normalised == original_dtype do
-        series
-      else
-        apply_series(series, :cast, [normalised])
-      end
-    else
-      dtype_error("cast/2", dtype, Shared.dtypes())
-    end
+  def cast(series, dtype) do
+    dtype |> Shared.normalise_dtype() |> maybe_cast_series(series)
+  end
+
+  defp maybe_cast_series({:error, dtype}, _), do: dtype_error("cast/2", dtype, Shared.dtypes())
+
+  defp maybe_cast_series(dtype, %Series{dtype: dtype} = series), do: series
+
+  defp maybe_cast_series(normalised, series), do: apply_series(series, :cast, [normalised])
+
+  @doc type: :element_wise
+  @spec nil_(series :: Series.t()) :: Series.t()
+  def nil_(series) do
+    cast(series, nil)
```

Comment:
I think I would call it `:null`, so we don't accidentally handle it as `nil`. And i don't think we need a `nil_` function, `cast(series, :null)` would be enough IMO.

Reply:
Also calling it `:null` would require fewer changes to the code, exactly becaude of stuff like `normalise_dtype` :D

Reply:
Thank you, fixed as :null - that resolved all failing test cases.